scripts/run-ios.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+set -euo pipefail
+set -m # needed to access jobs
+
+GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
+
+# 25 seconds is enough for metro to start, if it doesn't we have problems
+TIMEOUT=25
+TIMESPENT=1
+
+SIMULATOR=${1}
+
+# get our desired UUID
+UUID=$(xcrun simctl list devices | grep -E "$SIMULATOR \(" | head -n 1 | awk -F '[()]' '{print $2}')
+
+# get simulator status
+SIMULATOR_STATE=$(xcrun simctl list devices | grep -E "$SIMULATOR \(" | head -n 1 | awk '{print $NF}')
+
+# sometimes a simulator is already running, shut it down to avoid errors
+if [ "$SIMULATOR_STATE" != "(Shutdown)" ]; then
+    xcrun simctl shutdown "$UUID"
+fi
+
+# boot up iOS for simulator
+xcrun simctl boot "$UUID"
+
+# start the simulator
+open -a Simulator --args -CurrentDeviceUDID "$UUID"
+
+#iOS build of debug scheme
+xcodebuild -workspace "ios/StatusIm.xcworkspace" -configuration Debug -scheme StatusIm -destination id="$UUID" | xcbeautify
+
+# Ensure log file exists
+touch metro-server-logs.log
+
+# start metro and go to background
+nohup "${GIT_ROOT}"/scripts/start-react-native.sh 2>&1 | tee metro-server-logs.log &
+
+METRO_JOB=$(jobs | grep "metro")
+if [ $? -eq 0 ]; then
+    METRO_ID=$(echo "$METRO_JOB" | awk '{print $1}' | tr -d '[]+' )
+else
+    echo "Metro server background information not found."
+    exit 1
+fi
+
+cleanupMetro() {
+  echo "Performing cleanup actions..."
+  rm metro-server-logs.log
+  if [[ -n "${METRO_JOB:-}" ]]; then
+    METRO_ID=$(echo "$METRO_JOB" | awk '{print $1}' | tr -d '[]+')
+    echo "Killing Metro server with job ID $METRO_ID"
+    kill %"$METRO_ID"
+  fi
+  echo "Cleanup complete."
+}
+
+echo "Waiting for Metro server to be ready..."
+while ! grep -q "Welcome to Metro" metro-server-logs.log; do
+  echo -n "."
+
+  ((TIMESPENT++))
+  if ((TIMESPENT > TIMEOUT)); then
+    echo "metro server timed out, exiting"
+    cleanupMetro
```

Comment:
Using a `trap` will be more robust, since it will cover other ways this script can exit. Some examples:

https://github.com/status-im/status-mobile/blob/5fdaf0054dffc3d3dcc20f46d52f2163ad92b0ba/nix/scripts/build.sh#L10-L29
https://github.com/status-im/status-mobile/blob/5fdaf0054dffc3d3dcc20f46d52f2163ad92b0ba/scripts/build-android.sh#L31-L32

Reply:
addressed here : https://github.com/status-im/status-mobile/pull/18784/commits/d6a19d75da1010ab9e33dd395ba4c6e45e7456e6